packages/drivers/odsp-driver/src/opsCaching.ts

```diff
@@ -65,7 +65,14 @@ export class OpsCache {
 
 	public flushOps() {
 		for (const [key, value] of this.batches) {
-			if (value === null || !value.dirty) {
+			// Don't flush if the batch has no ops, already flushed or has empty slots at both beginning and end.
+			if (
+				value === null ||
+				!value.dirty ||
+				value.batchData.length === 0 ||
+				(value.batchData[0] === undefined &&
+					value.batchData[value.batchData.length - 1] === undefined)
```

Comment:
Do we have tests for each of these condition to test each codepath? It's probably easy to just have different types of mockData to execute each condition.

Reply:
Yes we already have tests related to ops caching but 1 of them failed which I expected, fixed that and added couple of more tests.

Reply:
Jatin, just so I can understand it, what kind of scenario would match this condition: 
(value.batchData[0] === undefined &&
					value.batchData[value.batchData.length - 1] === undefined  ?

Reply:
So we push ops in opsCache based on timer too. So lets say we have batches of size 100 and we started getting ops from 30 (for ex lets say we loaded from seq number 29), then we timer out and we only got ops from 30-80 by then, in that case we would not want to flush those ops as the batch is not filled yet.